javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -288,84 +1003,200 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
-        for (java.util.Map.Entry<? extends K, ? extends V> entry : entries) {
-            tree = tree.insert(Tuple.of(entry.getKey(), entry.getValue()));
-        }
-        return tree.isEmpty() ? empty(keyComparator) : new TreeMap<>(tree);
+    public static <K, V> TreeMap<K, V> ofEntries(Comparator<? super K> keyComparator, java.util.Map.Entry<? extends K, ? extends V> e1, java.util.Map.Entry<? extends K, ? extends V> e2, java.util.Map.Entry<? extends K, ? extends V> e3) {
+        Objects.requireNonNull(e3, "The 3rd entry is null");
+        return TreeMap.<K, V> ofEntries(keyComparator, e1, e2).put(e3.getKey(), e3.getValue());
     }
 
     /**
-     * Creates a {@code TreeMap} of the given entries.
+     * Creates a {@code TreeMap} of given entries.
      *
-     * @param <K>     The key type
-     * @param <V>     The value type
-     * @param entries Map entries
-     * @return A new TreeMap containing the given entries.
+     * @param <K> The key type
+     * @param <V> The value type
+     * @param e1  The 1st entry
+     * @param e2  The 2nd entry
+     * @param e3  The 3rd entry
+     * @param e4  The 4th entry
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @return A new TreeMap containing the given entry.
      */
-    public static <K extends Comparable<? super K>, V> TreeMap<K, V> ofEntries(Iterable<? extends Tuple2<? extends K, ? extends V>> entries) {
-        return ofEntries((Comparator<? super K> & Serializable) K::compareTo, entries);
+    public static <K, V> TreeMap<K, V> ofEntries(Comparator<? super K> keyComparator, java.util.Map.Entry<? extends K, ? extends V> e1, java.util.Map.Entry<? extends K, ? extends V> e2, java.util.Map.Entry<? extends K, ? extends V> e3, java.util.Map.Entry<? extends K, ? extends V> e4) {
+        Objects.requireNonNull(e4, "The 4th entry is null");
+        return TreeMap.<K, V> ofEntries(keyComparator, e1, e2, e3).put(e4.getKey(), e4.getValue());
+    }
+
+    /**
+     * Creates a {@code TreeMap} of given entries.
+     *
+     * @param <K> The key type
+     * @param <V> The value type
+     * @param e1  The 1st entry
+     * @param e2  The 2nd entry
+     * @param e3  The 3rd entry
+     * @param e4  The 4th entry
+     * @param e5  The 5th entry
+     * @param keyComparator The comparator used to sort the entries by their key.
+     * @return A new TreeMap containing the given entry.
+     */
+    public static <K, V> TreeMap<K, V> ofEntries(Comparator<? super K> keyComparator, java.util.Map.Entry<? extends K, ? extends V> e1, java.util.Map.Entry<? extends K, ? extends V> e2, java.util.Map.Entry<? extends K, ? extends V> e3, java.util.Map.Entry<? extends K, ? extends V> e4, java.util.Map.Entry<? extends K, ? extends V> e5) {
+        Objects.requireNonNull(e5, "The 5th entry is null");
+        return TreeMap.<K, V> ofEntries(keyComparator, e1, e2, e3, e4).put(e5.getKey(), e5.getValue());
```

Comment:
I think you could create a `t` here also, that creates a `Tuple` from an `Entry`